rclpy/test/test_node.py

```diff
@@ -838,6 +838,162 @@ def test_node_set_parameters_rejection(self):
+                ParameterDescriptor()
+            ),
+            (
+                'non_reject',
+                True,
+                ParameterDescriptor()
+            ),
+            (
+                'non_reject',
+                True,
+                ParameterDescriptor()
+            )
+        ]
+
+        self.node.declare_parameters('', reject_list_parameter_tuple1)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_1)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_2)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_3)
+
+        result = self.node.set_parameters(
+            [
+                Parameter(
+                    name=reject_list_parameter_tuple1[0][0],
+                    value=reject_list_parameter_tuple1[0][1]
+                ),
+                Parameter(
+                    name=reject_list_parameter_tuple1[1][0],
+                    value=reject_list_parameter_tuple1[1][1]
+                ),
+                Parameter(
+                    name=reject_list_parameter_tuple1[2][0],
+                    value=reject_list_parameter_tuple1[2][1]
+                )
+            ]
+        )
+        self.assertIsInstance(result, list)
+        self.assertIsInstance(result[0], SetParametersResult)
+        self.assertFalse(result[0].successful)
+
+        # Adding more parameters.
+        success_parameters_list_tuple = [
+            (
+                's1',
+                True,
+                ParameterDescriptor()
+            ),
+            (
+                's2',
+                True,
+                ParameterDescriptor()
+            ),
+            (
+                's3',
+                True,
+                ParameterDescriptor()
+            )
+        ]
+        self.node.declare_parameters('', success_parameters_list_tuple)
+        result = self.node._set_parameters(
+            [
+                Parameter(
+                    name=success_parameters_list_tuple[0][0],
+                    value=success_parameters_list_tuple[0][1]
+                ),
+                Parameter(
+                    name=success_parameters_list_tuple[1][0],
+                    value=success_parameters_list_tuple[1][1]
+                ),
+                Parameter(
+                    name=success_parameters_list_tuple[2][0],
+                    value=success_parameters_list_tuple[2][1]
+                )
+            ]
+        )
+        self.assertIsInstance(result, list)
+        self.assertIsInstance(result[0], SetParametersResult)
+        self.assertTrue(result[0].successful)
+
+    def test_node_add_on_set_paramter_callback(self):
+        # Add callbacks to the list of callbacks.
+
+        callbacks = [
+            self.reject_parameter_callback,
+            self.reject_parameter_callback_1,
+            self.reject_parameter_callback_2,
+            self.reject_parameter_callback_3
+        ]
```

Comment:
Why does this test need to add 4 different callback? Two should be sufficient. And it would be good to test that they are before not already in the list of callback.

Reply:
Should I change that to 2 callbacks only?
@dirk-thomas sir

Reply:
@dirk-thomas @allenh1 sir why is the jenkins not testing my code..did I commit a mistake somewhere?

Reply:
> why is the jenkins not testing my code..did I commit a mistake somewhere?

It looks like Travis CI doesn't run on rclpy anymore? @dirk-thomas would need to confirm that.

There is a Jenkins instance that is triggered manually.

@suab321321 Please refrain from force pushing to this branch -- it makes the review process much harder to follow. Instead, please push single commits as we go (since this PR will be squashed if it is to be merged).

Reply:
Testing is not done via Travis CI but via the ROS Jenkins server (build.ros2.org),

With the recent Eloquent release this repository has an Eloquent specific branch now. We only have Eloquent specific CI - CI for master / Foxy will only we available in the upcoming week(s).

Manually triggered builds are still possible on ci.ros2.org. We can trigger a set of builds once the PR is ready and no comments are pending.

Reply:
@allenh1 @dirk-thomas okay sir onto it :)

Reply:
@suab321321 there's still 4 callbacks here -- please reduce to two as Dirk suggested.